Facepunch.Steamworks/Structs/LobbyQuery.cs

```diff
@@ -43,6 +43,27 @@ public LobbyQuery FilterDistanceWorldwide()
 		}
 		#endregion
 
+		#region String key/value filter
+		internal MatchMakingKeyValuePair_t? stringFilter;
+
+		/// <summary>
+		/// Filter by specified key/value pair; string parameters
+		/// </summary>
+		public LobbyQuery FilterStringKeyValue( string nKey, string nValue )
+		{
+			if ( !string.IsNullOrEmpty(nKey) && nKey.Length <= SteamMatchmaking.MaxLobbyKeyLength )
```

Comment:
Seems like the right thing here is to throw exceptions about the arguments being wrong instead of just ignoring it

Reply:
Roger. I'll try to keep an eye out for that in the future. Is there any sort of universal OnException Action I should be invoking? I only see the OnCallbackException one in SteamClient

Reply:
No I'd just throw an System.ArgumentException.

The OnCallbackException is usually only called during async calls.

Reply:
Ah ok. Also one last question; throwing an exception essentially terminates the method at that point, right? Which would prevent the method from returning "this" on an exception, which would break the ability to call a method chain/fluent LobbyQuery interface, from my understanding, if another method came after. What would be the best way to approach that? If the public FilterStringKeyValue contained a try/catch/finally that tried an internal method containing this logic, caught the exceptions, then finally returned "this," the exceptions wouldn't bubble up to the level of code referencing the public method, or would it? I'm still trying to make sense of exception handling in traditional development situations.

Reply:
Exception are for the end user to handle. In this situation throwing an exception instead of doing nothing just tells the coder that they're calling the function wrong.. so the most likely reaction them is that they'd prevent it being called wrong rather than try to handle the exception - but the option is there for them too.

Reply:
Got it. Thank you for the perspective!